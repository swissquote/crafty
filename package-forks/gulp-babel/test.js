const { Transform } = require("node:stream");
const { test } = require("node:test");
const { expect } = require("expect");
const path = require("path");
const Vinyl = require("vinyl");
const sourcemap = require("vinyl-sourcemap");
const babel = require(".");

test("should transpile with Babel", () => {
  return new Promise(done => {
    const stream = babel({
      plugins: ["@babel/transform-block-scoping"]
    });

    stream.on("data", file => {
      expect(/var foo/.test(file.contents.toString())).toBe(true);
      expect(file.relative).toBe("fixture.js");
    });

    stream.on("end", done);

    stream.write(
      new Vinyl({
        cwd: __dirname,
        base: path.join(__dirname, "fixture"),
        path: path.join(__dirname, "fixture/fixture.jsx"),
        contents: Buffer.from("let foo;")
      })
    );

    stream.end();
  });
});

function initSourceMap() {
  return new Transform({
    objectMode: true,
    transform(file, enc, callback) {
      sourcemap.add(file, (sourcemapErr, updatedFile) => {
        if (sourcemapErr) {
          callback(sourcemapErr);
          return;
        }

        callback(null, updatedFile);
      });
    }
  });
}

function writeSourceMap() {
  return new Transform({
    objectMode: true,
    transform(file, enc, callback) {
      var self = this;

      sourcemap.write(file, (sourcemapErr, updatedFile, sourcemapFile) => {
        if (sourcemapErr) {
          callback(sourcemapErr);
          return;
        }

        self.push(updatedFile);
        if (sourcemapFile) {
          self.push(sourcemapFile);
        }

        callback();
      });
    }
  });
}

test("should generate source maps", () => {
  return new Promise(done => {
    const init = initSourceMap();
    const write = writeSourceMap();

    init
      .pipe(
        babel({
          plugins: ["@babel/transform-arrow-functions"]
        })
      )
      .pipe(write);

    write.on("data", file => {
      expect(file.sourceMap.sources).toEqual(["fixture.es2015"]);
      expect(file.sourceMap.file).toBe("fixture.js");
      const contents = file.contents.toString();
      expect(/function/.test(contents)).toBe(true);
      expect(/sourceMappingURL/.test(contents)).toBe(true);
      done();
    });

    init.write(
      new Vinyl({
        cwd: __dirname,
        base: path.join(__dirname, "fixture"),
        path: path.join(__dirname, "fixture/fixture.es2015"),
        contents: Buffer.from("[].map(v => v + 1)"),
        sourceMap: ""
      })
    );

    init.end();
  });
});

test("should generate source maps for file in nested folder", () => {
  return new Promise(done => {
    const init = initSourceMap();
    const write = writeSourceMap();
    init
      .pipe(
        babel({
          plugins: ["@babel/transform-arrow-functions"]
        })
      )
      .pipe(write);

    write.on("data", file => {
      expect(file.sourceMap.sources).toEqual(["nested/fixture.es2015"]);
      expect(file.sourceMap.file).toBe("nested/fixture.js");
      const contents = file.contents.toString();
      expect(/function/.test(contents)).toBe(true);
      expect(/sourceMappingURL/.test(contents)).toBe(true);
      done();
    });

    init.write(
      new Vinyl({
        cwd: __dirname,
        base: path.join(__dirname, "fixture"),
        path: path.join(__dirname, "fixture/nested/fixture.es2015"),
        contents: Buffer.from("[].map(v => v + 1)"),
        sourceMap: ""
      })
    );

    init.end();
  });
});

test("should pass the result of transform().metadata in file.babel", () => {
  return new Promise(done => {
    const stream = babel({
      plugins: [
        {
          post(file) {
            file.metadata.test = "metadata";
          }
        }
      ]
    });

    stream.on("data", file => {
      expect(file.babel).toEqual({ test: "metadata" });
    });

    stream.on("end", done);

    stream.write(
      new Vinyl({
        cwd: __dirname,
        base: path.join(__dirname, "fixture"),
        path: path.join(__dirname, "fixture/fixture.js"),
        contents: Buffer.from("class MyClass {};")
      })
    );

    stream.end();
  });
});

test("should not rename ignored files", () => {
  return new Promise(done => {
    const stream = babel({
      ignore: [/fixture/]
    });

    const inputFile = {
      cwd: __dirname
    };

    inputFile.base = path.join(inputFile.cwd, "fixture");
    inputFile.basename = "fixture.jsx";
    inputFile.path = path.join(inputFile.base, inputFile.basename);
    inputFile.contents = Buffer.from(";");

    stream
      .on("data", file => {
        expect(file.relative).toBe(inputFile.basename);
      })
      .on("end", done)
      .end(new Vinyl(inputFile));
  });
});

test("should not rename files without an extension", () => {
  return new Promise(done => {
    const stream = babel();

    const inputFile = {
      cwd: __dirname
    };

    inputFile.base = path.join(inputFile.cwd, "bin");
    inputFile.basename = "app";
    inputFile.path = path.join(inputFile.base, inputFile.basename);
    inputFile.contents = Buffer.from(";");

    stream
      .on("data", file => {
        expect(file.relative).toBe(inputFile.basename);
      })
      .on("end", done)
      .end(new Vinyl(inputFile));
  });
});
