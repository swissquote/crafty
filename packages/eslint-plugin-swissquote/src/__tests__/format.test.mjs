import test from "node:test";
import { expect } from "expect";
import initSnapshot from "@onigoetz/ntr-expect-snapshot";

import { prepareESLint } from "../../test_utils.js";

initSnapshot(import.meta.url);

const lint = prepareESLint("format");

test("Gives no warning on correct code", async t => {
  const result = await lint(
    `
module.exports = function initJS(gulp, config, watchers) {
  const js = config.js,
    jsTasks = [];

  for (const name in js) {
    if (!js.hasOwnProperty(name)) {
      continue;
    }

    const taskName = \`js_\${name}\`;

    if (!compileWithWebpack(js[name])) {
      gulp.task(taskName, jsTaskES5(gulp, config, watchers, js[name]));
      watchers.add(js[name].watch || js[name].source, taskName);
    }

    jsTasks.push(taskName);
  }

  gulp.task("js", jsTasks);

  return ["js"];
};
`
  );

  expect(result.messages).toMatchSnapshot();
  expect(result.warningCount).toEqual(0);
  expect(result.errorCount).toEqual(0);
});

test("Fails on badly formatted code", async t => {
  const result = await lint(
    `
module.exports = function initJS(gulp, config, watchers) {
  const js = config.js,
     jsTasks = [];

  for (const name in js) {
    if (!js.hasOwnProperty(name)) {
      continue;
      }

    const taskName = \`js_\${name}\`;

    if (!compileWithWebpack(js[name])) {
      gulp.task(taskName, jsTaskES5(gulp, config, watchers, js[name]));
        watchers.add(js[name].watch || js[name].source, taskName);
    }

    jsTasks.push( taskName );
  }

  gulp.task("js", jsTasks);

  return ["js"];
};
`
  );

  expect(result.messages).toMatchSnapshot();
  expect(result.warningCount).toEqual(0);
  expect(result.errorCount).toEqual(4);
});
