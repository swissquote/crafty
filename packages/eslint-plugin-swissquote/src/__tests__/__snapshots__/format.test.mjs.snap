// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fails on badly formatted code//0`] = `Array [
  Object {
    "column": 5,
    "endColumn": 6,
    "endLine": 3,
    "fix": Object {
      "range": Array [
        87,
        88,
      ],
      "text": "",
    },
    "line": 3,
    "message": "Delete \`·\`",
    "messageId": "delete",
    "nodeType": null,
    "ruleId": "@swissquote/swissquote/prettier/prettier",
    "severity": 2,
  },
  Object {
    "column": 1,
    "endColumn": 3,
    "endLine": 8,
    "fix": Object {
      "range": Array [
        182,
        184,
      ],
      "text": "",
    },
    "line": 8,
    "message": "Delete \`··\`",
    "messageId": "delete",
    "nodeType": null,
    "ruleId": "@swissquote/swissquote/prettier/prettier",
    "severity": 2,
  },
  Object {
    "column": 1,
    "endColumn": 9,
    "endLine": 14,
    "fix": Object {
      "range": Array [
        340,
        348,
      ],
      "text": "      ",
    },
    "line": 14,
    "message": "Replace \`········\` with \`······\`",
    "messageId": "replace",
    "nodeType": null,
    "ruleId": "@swissquote/swissquote/prettier/prettier",
    "severity": 2,
  },
  Object {
    "column": 18,
    "endColumn": 28,
    "endLine": 17,
    "fix": Object {
      "range": Array [
        431,
        441,
      ],
      "text": "taskName",
    },
    "line": 17,
    "message": "Replace \`·taskName·\` with \`taskName\`",
    "messageId": "replace",
    "nodeType": null,
    "ruleId": "@swissquote/swissquote/prettier/prettier",
    "severity": 2,
  },
]`;

exports[`Gives no warning on correct code//0`] = `Array []`;
